packages/inter-protocol/src/vaultFactory/vaultManager.js

```diff
@@ -607,25 +609,26 @@ export const prepareVaultManagerKit = (
             facets.helper.sendToReserve(accounting.overage, liqSeat, 'Minted');
 
             // return remaining funds to vaults before closing
-            const collateralPortion = makeRatioFromAmounts(
-              distributableCollateral,
-              totalCollateral,
-            );
 
             /** @type {import('@agoric/zoe/src/contractSupport/atomicTransfer.js').TransferPart[]} */
             const transfers = [];
             let collatRemaining = distributableCollateral;
 
-            for (const [vault, amounts] of vaultData.entries()) {
+            // iterate from best to worst, returning collateral until it has
+            // been exhausted. Vaults after that get nothing.
+            for (const [vault, amounts] of bestToWorst) {
+              const { collateralAmount: vCollat, debtAmount } = amounts;
               facets.helper.subtractPhantomInterest(
-                amounts.debtAmount,
+                debtAmount,
                 vault.getCurrentDebt(),
               );
-              if (collateralToDistribute) {
-                const collat = floorMultiplyBy(
-                  amounts.collateralAmount,
-                  collateralPortion,
-                );
+              const vaultDebt = floorMultiplyBy(debtAmount, debtPortion);
+              const collatPostDebt = AmountMath.subtract(vCollat, vaultDebt);
+              if (
+                collateralToDistribute &&
+                !AmountMath.isEmpty(collatRemaining)
+              ) {
+                const collat = AmountMath.min(collatRemaining, collatPostDebt);
```

Comment:
these two var names read as almost the same thing out of context (e.g. in a code review)

please rename `collatRemaining` to indicate it's out of what to distribute. I see why you don't mutate `distributableCollateral`, because it's not distributed until the `atomicRearrange` after the loop. It would be clearer if the mutating var indicated the "staging" role. Also it doesn't need "collate" because t's pretty clear from context that the thing being distributed is collateral so maybe,

`s/collatRemaining/leftToStage`